iphone/Maps/Classes/ClipboardMonitor.mm

```diff
@@ -0,0 +1,116 @@
+          NSString *message = [NSString stringWithFormat:@"Extracting coordinates from %@ using the anonymous proxy...", copiedText];
+            UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Please Wait" message:message preferredStyle:UIAlertControllerStyleAlert];
+            [window.rootViewController presentViewController:alertController animated:YES completion:nil];
+        });
+
+        // Make an API request
+      NSURL *apiURL = [NSURL URLWithString:[NSString stringWithFormat:privacyProxy, copiedText]];
+        NSURLSessionDataTask *task = [[NSURLSession sharedSession] dataTaskWithURL:apiURL completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+            dispatch_async(dispatch_get_main_queue(), ^{
+                [window.rootViewController dismissViewControllerAnimated:YES completion:nil];
+            });
+
+            if (error) {
+                NSLog(@"Proxy request error: %@", error);
+                NSString *errorMessage = @"Failed to extract coordinates from %url using an anonymous proxy. Please check your Internet connection";
+              errorMessage = [errorMessage stringByReplacingOccurrencesOfString:@"%url" withString:copiedText];
+                [self displayErrorMessage:errorMessage inWindow:window];
+            } else {
+                NSError *jsonError;
+                NSString *responseString = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+                NSLog(@"Proxy response:%@", responseString);
+                NSDictionary *responseDict = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:&jsonError];
+                if (jsonError) {
+                    NSLog(@"JSON Parsing Error: %@", jsonError);
+                    NSString *errorMessage = @"Can not extract coordinates from %url";
+                  errorMessage = [errorMessage stringByReplacingOccurrencesOfString:@"%url" withString:copiedText];
+                  [self displayErrorMessage:errorMessage inWindow:window];
+                } else {
+                    NSDictionary *urlDict = responseDict[@"url"];
+                    NSString *geoURL = urlDict[@"geo"];
+                    if (geoURL) {
+                        NSLog(@"Geo URL: %@", geoURL);
+                        dispatch_async(dispatch_get_main_queue(), ^{
+                            GetFramework().ShowMapForURL(geoURL.UTF8String);
+                        });
+                    }
+                }
+            }
+        }];
+        [task resume];
+    }
+}
+
+- (BOOL)isURL:(NSString *)text {
+    NSDataDetector *detector = [NSDataDetector dataDetectorWithTypes:NSTextCheckingTypeLink error:nil];
+    NSArray *matches = [detector matchesInString:text options:0 range:NSMakeRange(0, text.length)];
+
+    for (NSTextCheckingResult *match in matches) {
+        if (match.resultType == NSTextCheckingTypeLink) {
+            return YES;
```

Comment:
We use two spaces for indentation. We don't use bracers for one-liners.

Reply:
I have formatted the code.